src/Pact/Types/PactValue.hs

```diff
@@ -90,3 +106,211 @@ toPactValueLenient t = case toPactValue t of
+
+instance SizeOf Literal where
+  sizeOf (LString t) = (constructorCost 1) + (sizeOf t)
+  sizeOf (LInteger i) = (constructorCost 1) + (sizeOf i)
+  sizeOf (LDecimal d) = (constructorCost 1) + (sizeOf d)
+  sizeOf (LBool _) = (constructorCost 1) + 0
+  sizeOf (LTime ti) = (constructorCost 1) + (sizeOf ti)
+
+instance (SizeOf v) => SizeOf (Vector v) where
+  sizeOf v = vectorSize
+    where
+      vectorSize =
+        ((7 + vectorLength) * wordSize) + sizeOfContents
+      vectorLength = fromIntegral (V.length v)
+      sizeOfContents = V.foldl' (\acc pv -> acc + (sizeOf pv)) 0 v
+
+instance (SizeOf m) => SizeOf (ObjectMap m) where
+  -- newtype is free
+  sizeOf (ObjectMap m) = sizeOf m    
+
+instance (SizeOf k, SizeOf v) => SizeOf (M.Map k v) where
+  sizeOf m = mapSize
+    where
+      mapSize = (6 * mapLength * wordSize) + sizeOfKeys + sizeOfValues
+      mapLength = fromIntegral (M.size m)
+      sizeOfValues = M.foldl' (\acc pv -> acc + (sizeOf pv)) 0 m
+      sizeOfKeys = M.foldlWithKey' (\acc fk _ -> acc + (sizeOf fk)) 0 m
+
+instance (SizeOf p) => SizeOf (Guard p) where
+  sizeOf (GPact pg) = (constructorCost 1) + (sizeOf pg)
+  sizeOf (GKeySet ks) = (constructorCost 1) + (sizeOf ks)
+  sizeOf (GKeySetRef ksr) = (constructorCost 1) + (sizeOf ksr)
+  sizeOf (GModule mg) = (constructorCost 1) + (sizeOf mg)
+  sizeOf (GUser ug) = (constructorCost 1) + (sizeOf ug)
+
+
+instance SizeOf PactGuard where
+  sizeOf (PactGuard pid pn) =
+    (constructorCost 2) + (sizeOf pid) + (sizeOf pn)
+
+instance SizeOf KeySet where
+  sizeOf (KeySet pkArr ksPred) =
+    (constructorCost 2) + (sizeOf pkArr) + (sizeOf ksPred)
+
+instance SizeOf KeySetName where
+  -- newtype is free
+  sizeOf (KeySetName n) = sizeOf n
+
+instance SizeOf ModuleGuard where
+  sizeOf (ModuleGuard md n) =
+    (constructorCost 2) + (sizeOf md) + (sizeOf n)
+
+instance (SizeOf p) => SizeOf (UserGuard p) where
+  sizeOf (UserGuard n arr) =
+    (constructorCost 2) + (sizeOf n) + (sizeOf arr)
+
+instance SizeOf Name where
+  sizeOf (QName qn) = (constructorCost 1) + sizeOf qn
+  sizeOf (Name bn) = (constructorCost 1) + sizeOf bn
+
+instance SizeOf QualifiedName where
+  sizeOf (QualifiedName modName n i) =
+    (constructorCost 3) + (sizeOf modName) + (sizeOf n) + (sizeOf i)
+
+instance SizeOf BareName where
+  sizeOf (BareName n i) =
+    (constructorCost 2) + (sizeOf n) + (sizeOf i)
+
+instance SizeOf ModuleName where
+  sizeOf (ModuleName mn namespace) =
+    (constructorCost 2) + (sizeOf mn) + (sizeOf namespace)
+
+instance SizeOf Info where
+  sizeOf (Info may) = sizeOf may
+
+instance SizeOf Parsed where
+  sizeOf (Parsed d len) =
+    (constructorCost 2) + (sizeOf d) + (sizeOf len)
+
+instance SizeOf Delta where
+  -- Most of delta's arguments are unpacked (unboxed) Int64
+  -- Most unboxed types just take one word
+  -- Exceptions being Int64#, Word64#, and Double# which take 2 words on a 32-bit machine.
+  -- TODO - Assumes 64-bit machine  
```

Comment:
It's unlikely we'll ever care about 32-bit?

Reply:
Yep, we're assuming its 64 bit. Resolving this for now